### DISCUSSION TITLE
About maximum duration of segments

### DISCUSSION THREAD

**Discussion by enisimsar** [pyannote/pyannote-audio#863] [Q&A] [answered]
Hi all, 

Thanks for the great library. 

I am trying to segment audio with the `dia` pipeline.
``` python 
import torch
from pyannote.audio.features import RawAudio

pipeline = torch.hub.load("pyannote/pyannote-audio", "dia")

OWN_FILE = {"audio": input_file}

diarization = pipeline(OWN_FILE)
segments = diarization.for_json()["content"]
```

However, I need to specify minimum and maximum durations for segments. I checked the parameters of the parts of the pipeline and could not find a solution. Do you have any idea?

**Comment by hbredin** [comment, marked as answer]
There is no easy way to do that (though I am not sure why you would need to specify a maximum duration).

FYI, pyannote.audio 1.x will soon be superseded by [pyannote.audio 2.0](https://github.com/pyannote/pyannote-audio/tree/develop#whats-new-in-pyannoteaudio-20) with a [much better](https://github.com/pyannote/pyannote-audio/tree/develop#benchmark) speaker diarization pipeline, which has a `min_duration_on` hyper-parameter that could you use to set the minimum duration of speech turns.

**Reply by enisimsar** [reply]
Thanks a lot, it worked. I have tried some options for `min_duration_on`. Now, I can divide the longer segments. 

I will also change my code to 2.0.

**Reply by Yagna24** [reply]
Hi @enisimsar , thank you for asking this question, this also helps me. 
I have a doubt, where does the input audio .wav file processes first i.e. what .py file is responsible to fetch the audio file when we use 

```
OWN_FILE = {"audio": input_file}

diarization = pipeline(OWN_FILE)
```
